File: src/main/java/org/enlightenseries/DomainDictionary/domain/Domain.java
package org.enlightenseries.DomainDictionary.domain;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Domain {
  private Long id;
  private String name;
  private String format;
  private String description;
  private String existential;
}

File: src/main/web/app/models/domain.ts
import { RelatedDomain } from "./related-domain";

export class Domain {
  id: number;
  name: string;
  format: string;
  description: string;
  existential: string;
  created: Date;
  updated: Date;
  relatedDomains: RelatedDomain[];

  constructor(
    id?,
    name?,
    format?,
    description?,
    existential?,
    created?,
    updated?,
    relatedDomains?
  ) {
    this.id = id ? id : 1;
    this.name = name ? name : '';
    this.format = format ? format : '';
    this.description = description ? description : '';
    this.existential = existential ? existential : '';
    this.created = created ? created : new Date();
    this.updated = updated ? updated : new Date();
    this.relatedDomains = relatedDomains ? relatedDomains : [];
  }
}